### PR TITLE
Small updates to run_evaluator after adding dataset.data_type

### PR DESCRIPTION
**Pull request by nfcampos**

### CODE REVIEW

File: langchain/evaluation/run_evaluators/string_run_evaluator.py [hunk 1]
```diff
@@ -203,7 +206,12 @@ def map(self, example: Example) -> Dict[str, Any]:
                     " specify a reference_key."
                 )
             else:
-                return {"reference": list(example.outputs.values())[0]}
+                output = list(example.outputs.values())[0]
+                return {
+                    "reference": output
+                    if type(output) == str
+                    else self.serialize_chat_messages([output])
```
Comment (agola11) [inline langchain/evaluation/run_evaluators/string_run_evaluator.py:213]: It's a bit presumptuous to assume the example outputs will be either messages or strings right? They can be arbitrary objects? Think we should probably only handle those two cases in this mapper but throw a better error to the user if they try to use this with some crazy dataset

Comment (hinthornw) [reply langchain/evaluation/run_evaluators/string_run_evaluator.py:213]: Only valid for the LLM/Chat model data type datasets ya

Comment (nfcampos) [reply langchain/evaluation/run_evaluators/string_run_evaluator.py:213]: I can change the condition. But note the previous condition also assumes the value is stringified

Comment (nfcampos) [reply langchain/evaluation/run_evaluators/string_run_evaluator.py]: I've updated the condition

File: langchain/evaluation/run_evaluators/string_run_evaluator.py [hunk 2]
```diff
@@ -331,7 +341,7 @@ def from_model_and_evaluator(
                 " not yet implemented."
                 "Expected one of [BaseLanguageModel, Chain, Tool]."
             )
-        if reference_key is not None:
+        if reference_key is not None or isinstance(model, BaseLanguageModel):
```
Comment (hinthornw) [inline langchain/evaluation/run_evaluators/string_run_evaluator.py:344]: Hm `reference_key` here was also communicating whether an example was required at all. May be tricky

Comment (nfcampos) [reply langchain/evaluation/run_evaluators/string_run_evaluator.py:344]: That does not seem to be how reference_key is used inside StringExampleMapper, see

Screenshot 2023-07-07 at 21 09 53

You always pick a reference even if reference_key is None

Comment (nfcampos) [reply langchain/evaluation/run_evaluators/string_run_evaluator.py:344]: Here is a possible diff that maintains error message while not requiring reference_key for llm/chat examples

Screenshot 2023-07-07 at 21 13 40

Comment (hinthornw) [reply langchain/evaluation/run_evaluators/string_run_evaluator.py]: I like that. Will make the change.

Indeed within the example mapper it was fine, I was just raising an error in the run evaluator

**Comment by hinthornw** [comment]
I need to not use the messages_from_dict since we're doing the serialized format now

```

{'lc': 1, 'type': 'constructor', 'id': ['langchain', 'schema', 'messages', 'AIMessage'], 'kwargs': {'content': 'fake response'}}
````

**Comment by nfcampos** [comment]
@hinthornw you should support both. Something like

```py
from langchain.load.load import loads

if 'lc' in message_json:
  message = loads(message_json)
else:
  message = _map_from_dict(message_json)
```

**Comment by hinthornw** [comment]
That did the trick

**Pull request merged**